pkg/rpcclient/invoker/invoker.go

```diff
@@ -229,20 +229,22 @@ func iterateNext(rpc RPCSessions, sessionID uuid.UUID, iterator *result.Iterator
 	}
 
 	var result []stackitem.Item
-
 	if iterator.Values != nil {
 		count := min(num, len(iterator.Values))
 		result = append(result, iterator.Values[:count]...)
 		iterator.Values = iterator.Values[count:]
 		num -= count
 	}
 
-	if num > 0 && iterator.ID != nil {
+	if num > 0 && iterator.ID != nil && !iterator.Truncated {
```

Comment:
But "truncated" is only passed on iterator expansion while ID is only passed when true sessions are available.

Reply:
Starting from https://github.com/nspcc-dev/neo-go/pull/3827 `Truncated` may be passed even if "true" sessions are enabled (if `SessionExpansionEnabled` on RPC server).

Reply:
> But this means we should have iterator.Truncated here

Yes, that's true. At the same time,  we can't rely on `iterator.Truncated` if session expansion is disabled on RPC server because in this case `iterator.Truncated` is always `false`.

So it looks like `iterator.Truncated` field is useless when sessions are enabled. I see two alternatives:
1. Always set `iterator.Truncated` to `true` on RPC server side when sessions are enabled and session expansion is disabled.
2. Do not use `iterator.Truncated` on RPC client side at all when `iterator.ID != nil` (i.e. when sessions are enabled on RPC server side).

Reply:
To me the first option is a little bit preferable because there's no way to distinguish if session expansion enabled on RPC client side except via `getversion`. If we can cache this value on the client side (and in the iterator itself)  then the third option is:
```suggestion
	if num > 0 && iterator.ID != nil && (!iterator.SessionExpansionEnabled || iterator.Truncated) {
```

Reply:
> What's the problem we're trying to solve?

Checking `iterator.Truncated` may allow to save one additional RPC request in case if the number of requested elements is larger than the one stored in `iterator.Values`.

So this PR supposed to be an optimisation.